src/main/java/org/jenkinsci/test/acceptance/docker/fixtures/KerberosContainer.java

```diff
@@ -48,7 +48,15 @@
     static {
         String hostFqdn = null;
         try {
-            hostFqdn = InetAddress.getByAddress(new byte[]{127, 0, 0, 1}).getCanonicalHostName();
+            String jenkins_local_hostname = System.getenv("JENKINS_LOCAL_HOSTNAME");
+            if (jenkins_local_hostname != null && !jenkins_local_hostname.isEmpty()) {
+                // JENKINS_LOCAL_HOSTNAME is set...presumably to an IP address
+                // that is not 127.0.0.1.
+                hostFqdn = InetAddress.getLocalHost().getHostName();
```

Comment:
I am not sure I understand why does this work. It seem to depend on rather accidental (IMO) corelation between `InetAddress.getLocalHost().getHostName()` and `JENKINS_LOCAL_HOSTNAME`. If you point `JENKINS_LOCAL_HOSTNAME` to OpenStack Floating IP (which is my use-case), local network stack does not know a thing about that IP...

How about just using whatever hostname/IP we use to reach Jenkins under test (to ensure that kerberos client is talking to exact same hostname/IP as we configured on KDC side without any network magic involved)?